### PR TITLE
Add select organization dropdown #58 

### PR DESCRIPTION
**Pull request by andpii**
Just _frontend_ part. Please show me mistakes and tips, if any exists.
Also not sure where to place this component and what design to use.
Will start backend part, when organization functionality is finished.
**Using example**:
`<tasque-organizations-dropdown [availableOrganizations]="organizations"></tasque-organizations-dropdown>`
**Current design**:
- Passive state:
image
- Active state:
image

### CODE REVIEW

**Review by ismv — CHANGES_REQUESTED** [review]
You shouldn't get tied to something specific like organizations with such components, those should be customizable and reusable.
Also you may need to add some adjustments once this [PR](https://github.com/BinaryStudioAcademy/bsa-2022-tasque/pull/29) is merged

File: frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<mat-menu #organizationsMenu="matMenu">
+  <div *ngFor="let organization of availableOrganizations">
+    <button (click)="onClick(organization)" mat-menu-item>{{ organization }} Organization</button>
+  </div>
+</mat-menu>
+
+<button mat-button [matMenuTriggerFor]="organizationsMenu">
+  {{ selectedOrganization }} Organization
+  <svg width="11" height="7" viewBox="0 0 11 7" fill="" xmlns="http://www.w3.org/2000/svg">
```
Comment (Anatolii-Ivanov) [inline frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.html]: Move to a file and add as img

File: frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  styleUrls: ['./organizations-dropdown.component.sass']
+})
+export class OrganizationsDropdownComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+  private selectedOrganization$ = new BehaviorSubject<string>(this.selectedOrganization);
```
Comment (Anatolii-Ivanov) [inline frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts]: Move this to a separate service and use it from there

File: frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  set selectedOrganization(value: string) {
+    this.selectedOrganization$.next(value);
+    localStorage.setItem('selectedOrganization', value);
+  }
+
+  get selectedOrganization(): string {
+    return localStorage.getItem('selectedOrganization') ?? 'None';
+  }
```
Comment (Anatolii-Ivanov) [inline frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts]: And this parts

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

File: frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  ngOnInit(): void {
+    this.storageService.currentOrganizationId$.subscribe(
+      (result) => {
+        const searchedOrganization = this.availableOrganizations.find((x) => x.id == result);
+        if (searchedOrganization) {
+          this.currentOrganization = searchedOrganization;
+        }
+        else {
+          // eslint-disable-next-line no-console
```
Comment (ismv) [inline frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts]: add proper error handling

File: frontend/src/shared/shared.module.ts [hunk 5]
```diff
@@ -13,57 +13,63 @@ import { CreateOrganizationDialogComponent } from './components/create-organizat
...
-        LeftSidebarComponent,
-        DropdownComponent,
-        CreateProjectComponent,
-        CreateProjectDialogComponent,
-        DropdownComponent,
-        CreateOrganizationComponent,
-        CreateOrganizationDialogComponent,
-        LoginPageComponent,
-    ],
+        imports: [
```
Comment (ismv) [inline frontend/src/shared/shared.module.ts]: why do we need extra tab here?

**Review by ismv — CHANGES_REQUESTED** [review]

File: frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.html (hunk 1 above)
Comment (andpii) [reply frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.html]: Done

File: frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts (hunk 2 above)
Comment (andpii) [reply frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts]: Done

File: frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts (hunk 3 above)
Comment (andpii) [reply frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts]: Done

File: frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts (hunk 4 above)
Comment (andpii) [reply frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts]: Done

File: frontend/src/shared/shared.module.ts (hunk 5 above)
Comment (andpii) [reply frontend/src/shared/shared.module.ts]: Removed extra tabs

**Comment by andpew** [comment]
> 

Added universal tasque-menu-dropdown, but not sure how to use it in my case

**Comment by andpii** [comment]
> You shouldn't get tied to something specific like organizations with such components, those should be customizable and reusable. Also you may need to add some adjustments once this [PR](https://github.com/BinaryStudioAcademy/bsa-2022-tasque/pull/29) is merged

Added universal tasque-menu-dropdown, but not sure how to use it in my case

**Comment by ismv** [comment]
> > You shouldn't get tied to something specific like organizations with such components, those should be customizable and reusable. Also you may need to add some adjustments once this [PR](https://github.com/BinaryStudioAcademy/bsa-2022-tasque/pull/29) is merged
> 
> Added universal tasque-menu-dropdown, but not sure how to use it in my case

The idea behind customizability and reusability is that you can pass an array of entities to this component and the dropdown will work for those as well. For example, if we are working with projects we should be able to use this component too. We already have tasque-dropdown component, you can try reusing it for your case

**Comment by andpii** [comment]
> > > You shouldn't get tied to something specific like organizations with such components, those should be customizable and reusable. Also you may need to add some adjustments once this [PR](https://github.com/BinaryStudioAcademy/bsa-2022-tasque/pull/29) is merged
> > 
> > 
> > Added universal tasque-menu-dropdown, but not sure how to use it in my case
> 
> The idea behind customizability and reusability is that you can pass an array of entities to this component and the dropdown will work for those as well. For example, if we are working with projects we should be able to use this component too. We already have tasque-dropdown component, you can try reusing it for your case

I understand the main idea behind reusability. But in this case there are 2 different dropdowns: for content and for header. They differ one from another by styling and logic. So for header i created this second dropdown. And also i created component with organizations dropdown that will be moved to app src when header is ready. The main problem is that it's hard to use even dropdown that i made for header (while styles are almost the same) for organization dropdown, because the business logic behind it differs. 

So for now, i think, it should be pushed to dev, and when header is ready, i refactor this code.

File: frontend/src/shared/components/tasque-menu-dropdown/menu-dropdown.component.sass [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+button
+    &[aria-expanded="true"]
+        img
+            transform: rotate(180deg)
+
+span
+    color: colorscheme(white-100)
+
+::ng-deep .black-transparent
+    background: rgba(colorscheme(violet-400), 0.8)
```
Comment (Anatolii-Ivanov) [inline frontend/src/shared/components/tasque-menu-dropdown/menu-dropdown.component.sass]: Can you move it to global css file?

File: frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        (result) => {
+          if (result.body) {
+            this.currentOrganization = result.body;
+            this.availableOrganizations.push(this.currentOrganization);
+          }
+        }
+      );
+  }
+
+  @Input() public availableOrganizations: OrganizationModel[] = [];
```
Comment (Anatolii-Ivanov) [inline frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts]: Move it to the top of the file

First all variables, then constructor, ngOnInit, then functions

File: frontend/src/shared/components/tasque-menu-dropdown/menu-dropdown.component.sass (hunk 6 above)
Comment (andpii) [reply frontend/src/shared/components/tasque-menu-dropdown/menu-dropdown.component.sass]: Done

File: frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts (hunk 7 above)
Comment (andpii) [reply frontend/src/shared/components/organizations-dropdown/organizations-dropdown.component.ts]: Done

**Review by ismv — DISMISSED** [review]

**Review by ismv — APPROVED** [review]

**Review by Setup1337 — APPROVED** [review]

**Pull request merged**